services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -1180,9 +1182,29 @@ public boolean finalizeVirtualMachineProfile(VirtualMachineProfile profile, Depl
         s_logger.debug(String.format("Setting UseHttpsToUpload config on cmdline with [%s] value.", useHttpsToUpload));
         buf.append(" useHttpsToUpload=").append(useHttpsToUpload);
 
+        addSecondaryStorageServerAddressToBuffer(buf, secStore, vmName);
```

Comment:
@GutoVeronezi 
can you add the addresses of all image stores ?

Reply:
Sure, I'll take a look at this today.

Reply:
@weizhouapache @GutoVeronezi , though I think Wei's request is valid it seems to me to be out of scope for this PR and the issue it tries to solve . Shall we move this enhancement to a new PR?

Reply:
@DaanHoogland, I'm okay with both approaches; however, moving the enhancement to a new PR sounds better, as we could release the fix previously.

I'm doing another round of tests to post the results.